src/build.cc

```diff
@@ -172,10 +173,12 @@ void Plan::ScheduleWork(map<Edge*, Want>::iterator want_e) {
   Pool* pool = edge->pool();
   if (pool->ShouldDelayEdge()) {
     pool->DelayEdge(edge);
-    pool->RetrieveReadyEdges(&ready_);
+    EdgeSet new_edges;
+    pool->RetrieveReadyEdges(&new_edges);
+    ready_.push_multiple(new_edges.begin(), new_edges.end());
```

Comment:
nit: this seems inefficient, can we consider Pool::RetrieveReadyEdges() pushing directly into |ready_| instead? Since the EdgePriorityQueue is in `src/graph.h` now, that shouldn't be a problem.